src/Analyser/ConstantResolver.php

```diff
@@ -232,7 +232,7 @@ public function resolvePredefinedConstant(string $resolvedConstantName): ?Type
 			return IntegerRangeType::fromInterval(1, null);
 		}
 		if ($resolvedConstantName === '__COMPILER_HALT_OFFSET__') {
-			return IntegerRangeType::fromInterval(0, null);
+			return IntegerRangeType::fromInterval(1, null);
```

Comment:
Zero-byte PHP code itself cannot output `__COMPILER_HALT_OFFSET__`, so it is always `>=1`.

The minimum value is probably 48, but it looks unnatural, so leave it above 1.

* https://php-play.dev/?c=DwfgpgxgFg9gBAfQQYQPIFkAKBJAMgUQCUEAJAQVwBUFUAxWgZX2oQG4koBDAGwBcEIMALYAHAJbcwAJwAUASlZA&v=8.3&f=html